File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int tot=0;
        for(int i=0;i<nums.length;i++)
        {
            tot+=nums[i];
        }
        int sum=nums.length*(nums.length+1)/2;
        return sum-tot;
    }
}

/*class Solution {
    public int missingNumber(int[] nums) {
        Arrays.sort(nums);
        if(nums[0]!=0)
        return 0;
        boolean fl=true;
        int r=0;
        for(int i=0;i<nums.length-1;i++)
        {
            if(nums[i+1]-nums[i]!=1)
            {
                fl=false;
                r=i+1;
            }
        }
        if(fl==true)
        r=nums.length;
        return r;
    }
}*/

File: 0268-missing-number/0268-missing-number.c
int sum(int n)
{
    if(n==1)
    return 1;
    else 
    return n+sum(n-1);
}
int missingNumber(int* nums, int numsSize) {
    int i,s=0;
    for(i=0;i<numsSize;i++)
    {
        s+=nums[i];
    }
    int tot=sum(numsSize);
    int a=tot-s;
    return a;
}